### PR TITLE
add ContainerHostType on all message payloads to identify host type

### PR DESCRIPTION
**Pull request by shang-wang**
### What does this PR do?

We need to use additional flag to indicate the host type for container collection. This will help the backend pipeline to resolve host in different situations, especially when containers are running in "hostless" environment like fargate in ECS and EKS.

https://github.com/DataDog/agent-payload/pull/49

### Motivation

@DataDog/processes 

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [[constraint]]
   name = "github.com/DataDog/agent-payload"
-  version = "=4.29.0"
+  revision = "c192317fe7c4d65a0a0aafd768c134120637e512"
```
Comment (shang-wang) [inline Gopkg.toml:3]: I will change this to version once the agent-payload PR is merge and versioned.

File: Gopkg.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [[constraint]]
   name = "github.com/DataDog/agent-payload"
-  version = "=4.29.0"
+  revision = "79d10f8bc542e3981ee156abe761e0d1398d5ae7"
```
Comment (shang-wang) [inline Gopkg.toml:3]: I'll change this to version once the payload PR is merged and tagged.

**Review by kevinconaway — APPROVED** [review]
Do pod payloads need this too?

Its too bad that we need to add it to all of the. different message types. instead of there being a central place for it.

**Comment by shang-wang** [comment]
@kevinconaway I asked and they don't need that, in those environments there's no resource that they are collecting.

File: pkg/process/config/config.go [hunk 3]
```diff
@@ -351,6 +357,17 @@ func NewSystemProbeConfig(loggerName config.LoggerName, yamlPath string) (*Agent
 	return cfg, nil
 }
 
+// getContainerHostType uses the fargate library to detect container environment and returns the protobuf version of it
+func getContainerHostType() model.ContainerHostType {
+	switch fargate.GetOrchestrator() {
```
Comment (ahmed-mez) [inline pkg/process/config/config.go:362]: :+1:

**Pull request merged**